server/src/tests/integration/erc20.rs

```diff
@@ -174,3 +254,97 @@ pub async fn l2_erc20_balance_of(token: Address, account: Address, rpc_url: &str
 }
+
+pub async fn l2_erc20_allowance(
+    token: Address,
+    owner: Address,
+    spender: Address,
+    rpc_url: &str,
+) -> Result<U256> {
+    let provider = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .on_http(Url::parse(rpc_url)?);
+
+    let args = vec![
+        // The caller here does not matter because it is a view call.
+        MoveValue::Address(EVM_NATIVE_ADDRESS)
+            .simple_serialize()
+            .unwrap(),
+        MoveValue::Address(token.to_move_address())
+            .simple_serialize()
+            .unwrap(),
+        MoveValue::Address(owner.to_move_address())
+            .simple_serialize()
+            .unwrap(),
+        MoveValue::Address(spender.to_move_address())
+            .simple_serialize()
+            .unwrap(),
+    ];
+    let function_call = EntryFunction::new(
+        ModuleId::new(EVM_NATIVE_ADDRESS, ident_str!("erc20").into()),
+        ident_str!("allowance").into(),
+        Vec::new(),
+        args,
+    );
+    let tx_data = TransactionData::EntryFunction(function_call);
+    let data = bcs::to_bytes(&tx_data)?;
+    let eth_call_result = CallBuilder::new_raw(provider, data.into())
+        .to(EVM_NATIVE_ADDRESS.to_eth_address())
+        .call()
+        .await?;
+
+    let return_values = SerializedReturnValues {
+        mutable_reference_outputs: Vec::new(),
+        return_values: bcs::from_bytes(&eth_call_result)?,
+    };
+    let evm_result = extract_evm_result(return_values);
+
+    Ok(U256::from_be_slice(&evm_result.output))
+}
+
+pub async fn l2_erc20_approve(
+    from_wallet: &PrivateKeySigner,
+    token: Address,
+    spender: Address,
+    amount: U256,
+    rpc_url: &str,
+) -> Result<()> {
+    let from_address = from_wallet.address();
+    let provider = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .wallet(EthereumWallet::from(from_wallet.to_owned()))
+        .on_http(Url::parse(rpc_url)?);
+
+    let args = vec![
+        MoveValue::Address(from_address.to_move_address())
+            .simple_serialize()
+            .unwrap(),
+        MoveValue::Address(token.to_move_address())
+            .simple_serialize()
+            .unwrap(),
+        MoveValue::Address(spender.to_move_address())
+            .simple_serialize()
+            .unwrap(),
+        MoveValue::U256(amount.to_move_u256())
+            .simple_serialize()
+            .unwrap(),
+    ];
+    let function_call = EntryFunction::new(
+        ModuleId::new(EVM_NATIVE_ADDRESS, ident_str!("erc20").into()),
+        ident_str!("approve_entry").into(),
+        Vec::new(),
+        args,
+    );
+    let tx_data = TransactionData::EntryFunction(function_call);
+    let data = bcs::to_bytes(&tx_data)?;
+    let receipt = CallBuilder::new_raw(provider, data.into())
+        .to(EVM_NATIVE_ADDRESS.to_eth_address())
+        .send()
+        .await?
+        .get_receipt()
+        .await?;
+
+    println!("APPROVE RECEIPT {receipt:#?}");
```

Comment:
I believe this was a debug message. We can remove it.

Reply:
Done in https://github.com/MovedNetwork/op-move/pull/314/commits/1d225cd106a8628249a12288be4070c7e29ffef1